encoding/base64/src/base64.c

```diff
@@ -167,6 +167,29 @@ base64_decode(const char *str, void *data)
     return q - (unsigned char *) data;
 }
 
+int
+base64_decode_maxlen(const char *str, void *data, int len)
+{
+    const char *p;
+    unsigned char *q;
+
+    q = data;
+    for (p = str; *p && (*p == '=' || strchr(base64_chars, *p)); p += 4) {
+        if (q - (unsigned char *)data >= len) {
```

Comment:
It looks like function can write to memory beyond data + len which does not seem to be intentional.